src/main/java/org/apache/maven/plugins/dependency/ListRepositoriesMojo.java

```diff
@@ -104,8 +104,10 @@ public boolean visitLeave(DependencyNode node) {
                     .collect(Collectors.groupingBy(
                             repo -> repo.getMirroredRepositories().isEmpty()));
 
-            prepareRemoteRepositoriesList(message, repoGroupByMirrors.get(Boolean.TRUE));
-            prepareRemoteMirrorRepositoriesList(message, repoGroupByMirrors.get(Boolean.FALSE));
+            prepareRemoteRepositoriesList(
+                    message, repoGroupByMirrors.getOrDefault(Boolean.TRUE, Collections.emptyList()));
+            prepareRemoteMirrorRepositoriesList(
+                    message, repoGroupByMirrors.getOrDefault(Boolean.FALSE, Collections.emptyList()));
```

Comment:
We can also check in methods `prepareRemote*` if list is empty to avoid produce incomplete message.

Reply:
In case of an empty list is passed to the `prepareRemoteMirrorRepositoriesList()` it doesn't add anything into the `message`, so no check is necessary there. The `prepareRemoteRepositoriesList()` that is called before `prepareRemoteMirrorRepositoriesList()` always appends this title message:
```java
        message.append("Project remote repositories used by this build:").append(System.lineSeparator());
```
Not appending this title message in `prepareRemoteRepositoriesList()` when `prepareRemoteMirrorRepositoriesList()` prints something will look bad, IMHO.

I think a better solution would be checking `repositories.isEmpty()` before line 101 and if it is empty just print some other title message and exit from `doExecute()`. Would you like me to change this pull request this way? What title message for the case of empty `repositories` would you like to be printed? For example: "No remote repository is used by this build."

Reply:
check in line 101 - ok

> "No remote repository is used by this build." 

will be ok

Reply:
Done